tests/functional/spec/list_collector_variants_primary_person.spec.js

```diff
@@ -0,0 +1,121 @@
+import VariantBlockPage from "../generated_pages/list_collector_variants_primary_person/variant-block.page";
+import PrimaryPersonListCollectorPage from "../generated_pages/list_collector_variants_primary_person/primary-person-list-collector.page";
+import ListCollectorAddPage from "../generated_pages/list_collector_variants_primary_person/list-collector-add.page";
+import ListCollectorPage from "../generated_pages/list_collector_variants_primary_person/list-collector.page";
+import EditPersonPage from "../generated_pages/list_collector_variants_primary_person/list-collector-edit.page";
+import SubmitPage from "../generated_pages/list_collector_variants_primary_person/submit.page";
+import ThankYouPage from "../base_pages/thank-you.page.js";
+
+describe("List collector with variants primary person", () => {
+  describe("Given that person is lives in house", () => {
+    before("Load the survey", () => {
+      browser.openQuestionnaire("test_list_collector_variants_primary_person.json");
+      it("The user is asked questions about whether they live there", () => {
+        $(VariantBlockPage.yes()).click();
+        $(VariantBlockPage.submit()).click();
+        expect($(PrimaryPersonListCollectorPage.legend()).getText()).to.contain("Do you live here? (variant)");
+      });
+    });
+  });
+  describe("Given the user starts on the 'Do you like variant' question", () => {
+    before("Load the survey", () => {
+      browser.openQuestionnaire("test_list_collector_variants_primary_person.json");
+    });
+    it("When the user says that they do live there, then they are shown as the primary person", () => {
+      $(VariantBlockPage.yes()).click();
+      $(VariantBlockPage.submit()).click();
+      $(PrimaryPersonListCollectorPage.youLiveHereYes()).click();
+      $(PrimaryPersonListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue("John");
+      $(ListCollectorAddPage.lastName()).setValue("Doe");
+      $(ListCollectorAddPage.submit()).click();
+      expect($(ListCollectorPage.listLabel(1)).getText()).to.equal("John Doe (You)");
+    });
+    it("When the user adds another person, they are shown in the summary", () => {
```

Comment:
Is this the list collector summary rather than the summary? I also want to able to combine the GIven / When/Then. If I did this, it doesn't make much sense

"Given the user starts on the 'Do you like variant" "When the user adds another person, they are shown in the summary"

Reply:
yes, it is list collector summary, will update that.  These scenarios are based on one Given scenario right at the top so subsequent scenarios will have when and then.  These tests are in **continuation** so it should be read as
 "Given the user starts on the 'Do you like variant" 
     "When the user says that they do live there, then they are shown as the primary person" 
     "When the user adds another person, they are shown in the list collector summary"
  i don't see point of repeating Given when they originate from same step. as mentioned previously , these tests are following same pattern as https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/tests/functional/spec/list_collector_primary_person.spec.js